pkg/db/db.go

```diff
@@ -4309,3 +4344,443 @@ func (h *DBHandler) RunCustomMigrationEnvironments(ctx context.Context, getAllEn
 		return nil
 	})
 }
+
+type OverviewCacheRow struct {
+	EslId     EslId
+	Timestamp time.Time
+	Json      string
+}
+
+func (h *DBHandler) ReadLatestOverviewCache(ctx context.Context, transaction *sql.Tx) (*api.GetOverviewResponse, error) {
+	span, ctx := tracer.StartSpanFromContext(ctx, "readLatestOverviewCache")
+	defer span.Finish()
+	if h == nil {
+		return nil, fmt.Errorf("readLatestOverviewCache: DBHandler is nil")
+	}
+	if transaction == nil {
+		return nil, fmt.Errorf("readLatestOverviewCache: no transaction provided")
+	}
+
+	selectQuery := h.AdaptQuery(
+		"SELECT eslId, timestamp, json FROM overview_cache ORDER BY eslId DESC LIMIT 1;",
+	)
+
+	span.SetTag("query", selectQuery)
+	rows, err := transaction.QueryContext(
+		ctx,
+		selectQuery,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("could not query overview_cache table from DB. Error: %w", err)
+	}
+	defer func(rows *sql.Rows) {
+		err := rows.Close()
+		if err != nil {
+			logger.FromContext(ctx).Sugar().Warnf("row closing error: %v", err)
+		}
+	}(rows)
+	var row = &OverviewCacheRow{
+		EslId:     0,
+		Timestamp: time.Unix(0, 0),
+		Json:      "",
+	}
+	if rows.Next() {
+		err := rows.Scan(&row.EslId, &row.Timestamp, &row.Json)
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				return nil, nil
+			}
+			return nil, fmt.Errorf("error scanning overview_cache row from DB. Error: %w", err)
+		}
+	} else {
+		row = nil
+	}
+	err = closeRows(rows)
+	if err != nil {
+		return nil, err
+	}
+	if row != nil {
+		result := &api.GetOverviewResponse{
+			Branch:            "",
+			ManifestRepoUrl:   "",
+			Applications:      map[string]*api.Application{},
+			EnvironmentGroups: []*api.EnvironmentGroup{},
+			GitRevision:       "",
+		}
+		err = json.Unmarshal([]byte(row.Json), result)
+		if err != nil {
+			return nil, err
+		}
+		return result, nil
+	}
+	return nil, nil
+}
+
+func (h *DBHandler) WriteOverviewCache(ctx context.Context, transaction *sql.Tx, overviewResponse *api.GetOverviewResponse) error {
+	span, _ := tracer.StartSpanFromContext(ctx, "writeOverviewCache")
+	defer span.Finish()
+	if h == nil {
+		return fmt.Errorf("writeOverviewCache: DBHandler is nil")
+	}
+	if transaction == nil {
+		return fmt.Errorf("writeOverviewCache: no transaction provided")
+	}
+
+	insertQuery := h.AdaptQuery(
+		"INSERT INTO overview_cache (timestamp, Json) VALUES (?, ?);",
+	)
+	jsonResponse, err := json.Marshal(overviewResponse)
+	if err != nil {
+		return fmt.Errorf("could not marshal overview json data: %w", err)
+	}
+	span.SetTag("query", insertQuery)
```

Comment:
```suggestion
	span.SetTag("query", insertQuery)
	jsonResponse, err := json.Marshal(overviewResponse)
	if err != nil {
		return fmt.Errorf("could not marshal overview json data: %w", err)
	}
```
we should set the query tag as soon as we have it.